src/Authenticators/TOTPAuthenticator.php

```diff
@@ -26,47 +28,54 @@ class TOTPAuthenticator extends BootstrapMFAAuthenticator
      * a specific TOTP authenticator
      */
     private static $algorithm = 'sha1';
-
-
+    
     /**
-     * @param string $link
-     * @return \SilverStripe\Security\MemberAuthenticator\LoginHandler|static
+     * Get configured algorithm for TOTP Authenticator
+     *
+     * Must be one of: "sha1", "sha256", "sha512"
+     * If not specified or invalid, default to "sha1"
+     * @return string
      */
-    public function getLoginHandler($link)
+    public static function get_algorithm()
     {
-        return TOTPLoginHandler::create($link, $this);
+        $algorithm = self::config()->get('algorithm');
+
+        return in_array(strtolower($algorithm), ['sha1', 'sha256', 'sha512'])
+            ? $algorithm
+            : 'sha1';
```

Comment:
If you did go with the above, then this could also be a constant e.g. `self::ALGORITHM_DEFAULT`